test-suite/src/function/custom.rs

```diff
@@ -0,0 +1,127 @@
+use {
+    crate::*,
+    gluesql_core::{
+        executor::{AlterError, EvaluateError},
+        prelude::{Payload, PayloadVariable, Value::*},
+        translate::TranslateError,
+    },
+};
+
+test_case!(custom, async move {
+    let test_cases = [
+        (
+            "CREATE FUNCTION add_none()",
+            Err(TranslateError::UnsupportedEmptyFunctionBody.into()),
+        ),
+        (
+            "CREATE FUNCTION add_none() RETURN null",
+            Ok(Payload::Create),
+        ),
+        (
+            "CREATE FUNCTION add_zero(n INT) RETURN n",
+            Ok(Payload::Create),
+        ),
+        (
+            "CREATE FUNCTION add_zero(n INT) RETURN n",
+            Err(AlterError::FunctionAlreadyExists("add_zero".to_owned()).into()),
+        ),
+        (
+            "CREATE FUNCTION add_one (n INT, x INT DEFAULT 1) RETURN n + x",
+            Ok(Payload::Create),
+        ),
+        (
+            "CREATE FUNCTION add_two (n INT, x INT DEFAULT 1, y INT) RETURN n + x + y",
+            Err(AlterError::NonDefaultArgumentFollowsDefaultArgument.into()),
+        ),
+        (
+            "CREATE FUNCTION add_two (n INT, x INT DEFAULT 1, y INT DEFAULT 1) RETURN n + x + y",
+            Ok(Payload::Create),
+        ),
+        ("SELECT add_none() AS r", Ok(select_with_null!(r; Null))),
+        (
+            "SELECT add_one(1) AS r",
+            Ok(select!(
+                r
+                I64;
+                2
+            )),
+        ),
+        (
+            "SELECT add_one(1, 8) AS r",
+            Ok(select!(
+                r
+                I64;
+                9
+            )),
+        ),
+        (
+            "SELECT add_one(1, 2, 4)",
+            Err(EvaluateError::FunctionArgsLengthNotWithinRange {
+                name: "add_one".to_owned(),
+                expected_minimum: 1,
+                expected_maximum: 2,
+                found: 3,
+            }
+            .into()),
+        ),
+        (
+            "SELECT add_one()",
+            Err(EvaluateError::FunctionArgsLengthNotWithinRange {
+                name: "add_one".to_owned(),
+                expected_minimum: 1,
+                expected_maximum: 2,
+                found: 0,
+            }
+            .into()),
+        ),
+        (
+            "SELECT add_two(1, null, 2) as r",
+            Ok(select!(
+                r
+                I64;
+                4
+            )),
```

Comment:
default is only applied when the argument is not specified.
e.g.
```bash
gluesql> create table foo (id integer, c integer null default 1);
gluesql> insert into foo (id) values (100);
1 row inserted

gluesql> select * from foo;
| id  | c |
|-----|---|
| 100 | 1 |

gluesql> insert into foo (id, c) values (100, NULL);
1 row inserted

gluesql> select * from foo;
| id  | c    |
|-----|------|
| 100 | 1    |
| 100 | NULL |
```

So, the expected value would be `Value::Null`